engine/skiplist.hpp

```diff
@@ -341,6 +341,18 @@ class Skiplist : public PersistentList {
                          deleted_record, prev_record_lock);
   }
 
+  bool ValidateDLRecord(const DLRecord *record) {
+    // if record is not a valid dl record, then its prev maybe a invalid
+    // pointer, so we need to catch exception here
+    try {
+      DLRecord *prev = pmem_allocator_->offset2addr<DLRecord>(record->prev);
+      return prev->next == pmem_allocator_->addr2offset(record) &&
+             SkiplistId(record) == id_;
+    } catch (std::runtime_error &e) {
```

Comment:
What runtime_error will be catched? Does any function in this function throws an standard library exception?